front/lib/api/assistant/configuration.ts

```diff
@@ -200,37 +204,81 @@ export async function createAgentConfiguration(
     status,
     generation,
     action,
+    sId,
   }: {
     name: string;
     description: string;
     pictureUrl: string;
     status: AgentConfigurationStatus;
     generation: AgentGenerationConfigurationType | null;
     action: AgentActionConfigurationType | null;
+    version?: number;
+    sId?: string;
   }
 ): Promise<AgentConfigurationType> {
   const owner = auth.workspace();
   if (!owner) {
     throw new Error("Unexpected `auth` without `workspace`.");
   }
 
-  // Create Agent config
-  const agentConfig = await AgentConfiguration.create({
-    sId: generateModelSId(),
-    status: status,
-    name: name,
-    description: description,
-    pictureUrl: pictureUrl,
-    workspaceId: owner.id,
-    generationConfigurationId: generation?.id || null,
-    // We know here that the retrievalConfiguration is one that we created and not a "global
-    // virtual" one so we're good to set the foreign key.
-    retrievalConfigurationId: action?.id || null,
-  });
+  let version = 0;
+
+  const agentConfig = await front_sequelize.transaction(
+    async (t): Promise<AgentConfiguration> => {
+      if (sId) {
+        const latestVersion = await AgentConfiguration.max<
+          number | null,
+          AgentConfiguration
+        >("version", {
+          where: {
+            workspaceId: owner.id,
+            sId: sId,
+          },
+          transaction: t,
+        });
+
+        if (latestVersion !== null) {
+          version = latestVersion + 1;
+        }
+
+        await AgentConfiguration.update(
+          { status: "archived" },
+          {
+            where: {
+              sId: sId,
+              workspaceId: owner.id,
+            },
+            transaction: t,
+          }
+        );
+      }
```

Comment:
NIT: Could be like this to match the query in getAgentConfiguration
```js
const agent = await AgentConfiguration.findOne({
    where: {
      sId: agentId,
      workspaceId: owner.id,
    },
    order: [["version", "DESC"]],
    ],
    limit: 1,
  } , transaction: t );
await agent.update({ status: "archived" }, transaction: t}
```

or would that work? (I'm not sure)
```js
await AgentConfiguration.update(
    { status: 'archived' },
    {
      where: {
        sId: sId,
        workspaceId: owner.id,
        version: {
          [Op.eq]: Sequelize.literal(`(SELECT MAX(version) FROM AgentConfigurations WHERE sId = ${sId} AND workspaceId = ${owner.id})`)
        }
      },
      transaction: t
    }
  );

```

Reply:
```
const agent = await AgentConfiguration.findOne({
    where: {
      sId: agentId,
      workspaceId: owner.id,
    },
    order: [["version", "DESC"]],
    ],
    limit: 1,
  } , transaction: t );
await agent.update({ status: "archived" }, transaction: t}
```

This seems like more work no ? We'd need to fetch the whole config for nothing


For the second one.... I'm not a fan of the raw sql without type checking + I still need the actual version number so I can +1 it to create the new one

Reply:
Sure! Works as is!